.github/workflows/clang-tidy.yaml

```diff
@@ -25,6 +25,17 @@ jobs:
         spack external find
         spack install googletest
         spack install trilinos@master~mpi~epetra+basker
+    - name: Cache install yaml-cpp
+      id: cache-yaml-cpp
+      uses: actions/cache@v3
+      with:
+        path: ${{ github.workspace }}/spack
+        key: linux-yaml-cpp
+    - name: Install yaml-cpp
+      if: steps.cache-yaml-cpp.outputs.cache-hit != 'true'
+      run: |
+        source spack/share/spack/setup-env.sh
+        spack install yaml-cpp
```

Comment:
Once you push this in, I'm thinking of unifying it with the action above so that we only have one "Build Dependencies" step/cache.  Any thoughts/objections?

Reply:
No objections from me, feel free to take care of it when you work on the CI pipeline as discussed.